### DISCUSSION TITLE
How to delete all rows of all tables from database, or otherwise clear a database entirely?

### DISCUSSION THREAD

**Discussion by triallax** [simolus3/drift#3027] [Q&A] [answered]
I have a log out/account deletion feature in my app, and I want to delete the app's database when that happens.

**Comment by simolus3** [comment, marked as answer]
So the easiest way is to just close the database instance and then delete the database `File` that you're passing to `NativeDatabase`.

If it's easier to keep the database open, you can delete content from all tables like this:

```dart
await transaction(() async {
  for (final table in allTables) {
    await delete(table).go();
  }
});
```

**Reply by triallax** [reply]
Thanks, I imagine that such a transaction could leave at least traces of the rows behind? In that case it's probably best if I delete the database file.

Thanks for answering such a simple question by the way, sending massive appreciation your way for creating and maintaining the awesome Drift and being so helpful. :)
